Implement route support for http request metrics (#34)

jekkel opened the pull request:
fixes #34 

Companion PRs:
* required: https://github.com/eclipse-vertx/vert.x/pull/3532
* https://github.com/vert-x3/vertx-web/pull/1682
* https://github.com/vert-x3/vertx-dropwizard-metrics/pull/101

jekkel commented:
@jotak would you mind having a look? Both vert.x core and vert.x web PRs are merged already.

jotak commented:
Hi @jekkel , sorry I was on a break, I will take a look asap

jotak commented on src/main/java/io/vertx/micrometer/impl/VertxHttpServerMetrics.java:
```diff
@@ -28,7 +28,11 @@
 import io.vertx.micrometer.impl.meters.Gauges;
 import io.vertx.micrometer.impl.meters.Timers;
 
+import java.util.ArrayList;
```
Some of these imports are unused

jekkel replied:
Good find! Cleaning up.

jotak commented:
Some tests are failing in `VertxHttpClientServerMetricsTest` , function `shouldReportHttpServerMetrics`. Some of the failures are related to your changes, and some others weren't ; I've fixed the latter in master so you should be able to fix the former after rebasing from master.

Other than that, the PR looks good to me!

jekkel commented:
Thanks for the review! I rebased from master, fixed the test and the documentation. There has been a wrong metric name in the docs: `vertx_http_server_processingTime` but the source code uses `vertx_http_server_responseTime` so I fixed that when adding the new `route` label to the docs.

jekkel commented:
I wonder whether this also resolved #86 ?

jotak reviewed (approved):
LGTM, thank you @jekkel !

jotak commented:
> I wonder whether this also resolves #86 ?

Yes there was several issues discussed in #86 and indeed there was the problem of solving path label cardinality issue; with one solution being to configure some regex-based filtering (as described here https://github.com/vert-x3/vertx-micrometer-metrics/issues/86#issuecomment-556003656 ) ; and this PR now also provides a good alternative.
